cv-data-service-library/src/main/java/com/trihydro/library/helpers/JsonToJavaConverter.java

```diff
@@ -242,7 +242,17 @@ else if (reverse != null)
         }
 
         if (regionDirectionNode != null) {
-            region.setDirection(mapper.treeToValue(regionDirectionNode, String.class));
+            var direction = "";
+            Iterator<Map.Entry<String, JsonNode>> fields = regionDirectionNode.fields();
+            while (fields.hasNext()) {
+                Map.Entry<String, JsonNode> field = fields.next();
+                var directionBool = field.getValue().toString();
+                if ("true".equals(directionBool))
+                    direction += "1";
+                else
+                    direction += "0";
+            }
+            region.setDirection(direction);
```

Comment:
suggestion(non-blocking): use a StringBuilder for more efficient string construction in a loop, and use the built-in Boolean.parseBoolean to get the boolean value of a string
```suggestion
            StringBuilder directionBuilder = new StringBuilder();
            Iterator<Map.Entry<String, JsonNode>> fields = regionDirectionNode.fields();

            while (fields.hasNext()) {
                Map.Entry<String, JsonNode> field = fields.next();
                boolean directionBool= Boolean.parseBoolean(field.getValue().toString());
                directionBuilder.append(directionBool ? "1" : "0");
            }

            region.setDirection(directionBuilder.toString());
```

Reply:
Updated - thank you for this suggestion!